### DISCUSSION TITLE
How to prepare the desired OS image and flash it, with the bootloader? (and are they suppsed to be gzipped containers?)

### DISCUSSION THREAD

**Discussion by ambein** [johang/sd-card-images#151] [General]
The boot-loader and stable version images worked well, thank you. 

It is not clear what makes a workable image and what doesn't, so I can prepare my own install images. What worked consistently was flashing the first partition with U-Boot, but in the OS image partition only the barebones Debian or Manjaro installer would work -- why?

When the bootloader and OS package are not gz-archived steps I figured are:

> dd if=u-boot.img of=/dev/device conv=notrunc,fsync

> dd if=OS.img seek=next_partition_byte of=/dev/device conv=notrunc,fsync

boot.img is the U-boot bootloader (not sure if Tow-Boot or other boot loaders are possible).
OS.img would be the distro of interest
next_partition_byte is byte after the boot.img is written (given output of dd)
/dev/device is the mount point of the target flash drive (lsblk is your friend)

With respect to the gz archives, is there a particular way to prepare the images into that format, i.e. as a container (singularity no longer available)? For instance one the OS-distro images format was .xz.

My success was inconsistent, really only with flashing the boot loader in the first partition. How would I be able to add a desired distro to the next partition? Personally, I am interested in the net-install version or graphic installer of debian (bookworm) for instance instead?

I'd like to know how best to partition and flash the images myself!

**Comment by johang** [comment]
The bootloader loads the boot script `/boot/boot.scr` in the OS partition. This boot script then loads and starts the kernel. If the boot script is not in the OS partition the boot will likely fail.
